tests/examples/wibox/awidget/prompt/highlight.lua

```diff
@@ -0,0 +1,66 @@
+local parent    = ... --DOC_NO_USAGE --DOC_HIDE
+local wibox     = require( "wibox"     ) --DOC_HIDE
+local awful     = { prompt = require("awful.prompt") }--DOC_HIDE
+local beautiful = require( "beautiful" ) --DOC_HIDE
+
+    local amp = "&amp"..string.char(0x3B)
+    local quot = "&quot"..string.char(0x3B)
+
+    local atextbox = wibox.widget.textbox()
+
+    -- Create a shortcut function
+    local function echo_test()
+        awful.prompt.run {
+            prompt       = "<b>Echo: </b>",
+            text         = 'a_very "nice" $SHELL && command', --DOC_HIDE
+            bg_cursor    = "#ff0000",
+            -- To use the default `rc.lua` prompt:
+            --textbox      = mouse.screen.mypromptbox.widget,
+            textbox      = atextbox,
+            highlighter  = function(b, a)
+                -- Add a random marker to delimitate the cursor
+                local cmd = b.."ZZZZZ"..a
```

Comment:
Trading a mild hack for an obvious bug doesn't seems like such a good idea to me. Maybe I should simply replace `ZZZZZ` by a better name (like `ZZZCURSORZZZ`[1])?

[1] I prefer to limit it to a single char category (like upper caps) to avoid too many of the parser expressions breakage. `ZZZ` because it has the lowest risk of collision (beside a PIZZZA typo).